src/screens/shared/FarmSearch.js

```diff
@@ -0,0 +1,135 @@
+import React, { Component } from 'react';
+import Modal from 'react-modal';
+
+import '../../styles/FarmSearch.css';
+import FarmCard from './components/FarmCard';
+import NewFarmCard from './components/NewFarmCard';
+
+import SearchIcon from '../../assets/search-icon.png';
+import farmProfileDefaultCover from '../../assets/farmProfileCover.png';
+import farmProfileDefaultIcon from '../../assets/farmProfilePhoto.png';
+
+const ROOT_ELEMENT = '#root';
+Modal.setAppElement(ROOT_ELEMENT);
+
+class FarmSearch extends Component {
+  constructor(props) {
+    super(props);
+    this.state = {
+      showFarmModal: false
```

Comment:
All things related to modal should be under FarmCard instead of FarmSearch since each farm card should have a modal. And you should check if they are FoodHub in the FarmCard component (passed in as prop) to see if there should be a modal.

Reply:
done
![image](https://user-images.githubusercontent.com/43193441/101261595-8e337300-3706-11eb-96f5-ee43e22b35f4.png)
![image](https://user-images.githubusercontent.com/43193441/101261602-97244480-3706-11eb-9bdd-b671b210ce35.png)

Reply:
first part done. second part edited (modal no longer dependent on foodhub or gap). 

add a prop in the function. if user credentials is Admin or Restricted Admin.
Admin (NSEVP) = Full Page, Restricted Admin (KS) = Modals.